vms/platformvm/camino_service.go

```diff
@@ -481,6 +483,135 @@ func (s *CaminoService) RegisterNode(_ *http.Request, args *RegisterNodeArgs, re
 	return errs.Err
 }
 
+type SetValidatorStateArgs struct {
+	api.UserPass
+	api.JSONFromAddrs
+
+	Change platformapi.Owner `json:"change"`
+	NodeID ids.NodeID        `json:"nodeID"`
+	State  byte              `json:"state"`
+}
+
+// SetValidatorState issues a SetValidatorStateTx
+func (s *CaminoService) SetValidatorState(_ *http.Request, args *SetValidatorStateArgs, reply *api.JSONTxID) error {
+	s.vm.ctx.Log.Debug("Platform: SetValidatorState called")
+
+	// Parse the from addresses
+	fromAddrs, err := avax.ParseServiceAddresses(s.addrManager, args.From)
+	if err != nil {
+		return err
+	}
+
+	user, err := keystore.NewUserFromKeystore(s.vm.ctx.Keystore, args.Username, args.Password)
+	if err != nil {
+		return err
+	}
+
+	// Get the user's keys
+	privKeys, err := keystore.GetKeychain(user, fromAddrs)
+	if err != nil {
+		return fmt.Errorf("couldn't get addresses controlled by the user: %w", err)
+	}
+
+	if err := user.Close(); err != nil {
+		return err
+	}
+
+	if len(privKeys.Keys) == 0 {
+		return errNoKeys
+	}
+
+	change, err := s.getOutputOwner(&args.Change)
+	if err != nil {
+		return err
+	}
+
+	// Create the transaction
+	tx, err := s.vm.txBuilder.NewSetValidatorStateTx(
+		args.NodeID,
+		validator.State(args.State),
+		privKeys.Keys,
+		change,
+	)
+	if err != nil {
+		return fmt.Errorf("couldn't create tx: %w", err)
+	}
+
+	reply.TxID = tx.ID()
+
+	errs := wrappers.Errs{}
+	errs.Add(
+		err,
+		s.vm.Builder.AddUnverifiedTx(tx),
+	)
+
+	return errs.Err
+}
+
+// GetDeferredValidatorsArgs are the arguments for calling GetDeferredValidators
+type GetDeferredValidatorsArgs struct {
+	// Subnet we're getting the deferred validators of
+	// If omitted, defaults to primary network
+	SubnetID ids.ID `json:"subnetID"`
+	// NodeIDs of validators to request. If [NodeIDs]
+	// is empty, it fetches all deferred validators. If
+	// some requested nodeIDs are not deferred validators,
+	// they are omitted from the response.
+	NodeIDs []ids.NodeID `json:"nodeIDs"`
+}
+
+// GetDeferredValidatorsReply are the results from calling GetDeferredValidators.
+// Unlike GetCurrentValidatorsReply, each validator has a null delegator list.
+type GetDeferredValidatorsReply struct {
+	Validators []interface{} `json:"validators"`
+}
+
+// GetDeferredValidators returns the list of deferred validators
+func (s *CaminoService) GetDeferredValidators(_ *http.Request, args *GetDeferredValidatorsArgs, reply *GetDeferredValidatorsReply) error {
+	s.vm.ctx.Log.Debug("Platform: GetDeferredValidators called")
+
+	reply.Validators = []interface{}{}
+
+	// Create set of nodeIDs
+	nodeIDs := set.Set[ids.NodeID]{}
+	nodeIDs.Add(args.NodeIDs...)
+	includeAllNodes := nodeIDs.Len() == 0
```

Comment:
```suggestion
	getAllNodes := nodeIDs.Len() == 0
```
but do it as you prefer